builtin/nomad/jobspec/releaser.go

```diff
@@ -0,0 +1,577 @@
+	log hclog.Logger,
+	release *Release,
+	ui terminal.UI,
+) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	rm := r.resourceManager(log, nil)
+
+	// If we don't have resource state, this state is from an older version
+	// and we need to manually recreate it.
+	if release.ResourceState == nil {
+		rm.Resource(rmResourceJobName).SetState(&Resource_Job{
+			Name: rmResourceJobName,
+		})
+	} else {
+		// Load our set state
+		if err := rm.LoadState(release.ResourceState); err != nil {
+			return err
+		}
+	}
+
+	return rm.DestroyAll(sg, ui)
+}
+
+func (r *Releaser) Status(
+	ctx context.Context,
+	log hclog.Logger,
+	release *Release,
+	ui terminal.UI,
+) (*sdk.StatusReport, error) {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	rm := r.resourceManager(log, nil)
+
+	// If we don't have resource state, this state is from an older version
+	// and we need to manually recreate it.
+	if release.ResourceState == nil {
+		rm.Resource(rmResourceJobName).SetState(&Resource_Job{
+			Name: rmResourceJobName,
+		})
+	} else {
+		// Load our set state
+		if err := rm.LoadState(release.ResourceState); err != nil {
+			return nil, err
+		}
+	}
+
+	step := sg.Add("Getting status of Nomad release...")
+	defer step.Abort()
+
+	resources, err := rm.StatusAll(ctx, log, sg, ui)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "resource manager failed to generate resource statuses: %s", err)
+	}
+
+	if len(resources) == 0 {
+		// This shouldn't happen - the status func for the releaser should always return a resource or an error.
+		return nil, status.Errorf(codes.Internal, "no resources generated for release - cannot determine status.")
+	}
+
+	var jobResource *sdk.StatusReport_Resource
+	for _, r := range resources {
+		if r.Type == "job" {
+			jobResource = r
+			break
+		}
+	}
+	if jobResource == nil {
+		return nil, status.Errorf(codes.Internal, "no job resource found - cannot determine overall health")
+	}
+
+	// Create our status report
+	result := sdk.StatusReport{
+		External:      true,
+		GeneratedTime: timestamppb.Now(),
+		Resources:     resources,
+		Health:        jobResource.Health,
+		HealthMessage: jobResource.HealthMessage,
+	}
+
+	// update output based on main health state
+	step.Update("Finished building report for Nomad platform")
+	step.Done()
+
+	// NOTE(briancain): Replace ui.Status with StepGroups once this bug
+	// has been fixed: https://github.com/hashicorp/waypoint/issues/1536
+	st := ui.Status()
+	defer st.Close()
+
+	// More UI detail for non-ready resources
+	for _, resource := range result.Resources {
+		if resource.Health != sdk.StatusReport_READY {
+			st.Step(terminal.StatusWarn, fmt.Sprintf("Resource %q is reporting %q", resource.Name, resource.Health.String()))
+		}
+	}
+
+	return &result, nil
+}
+
+// ReleaserConfig is the configuration structure for the Releaser.
+type ReleaserConfig struct {
+	Groups         []string `hcl:"groups,optional"`
+	FailDeployment bool     `hcl:"fail_deployment,optional"`
```

Comment:
Variable comments here would be excellent! Could be a copy of the doc comments defined below.